app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+
+const Greeting = () => {
+  const [greeting, setGreeting] = useState('');
+
+  useEffect(() => {
+    const fetchGreeting = async () => {
+      const response = await fetch('/api/greetings/random_batch?count=1');
+      const data = await response.json();
+      setGreeting(data.greetings[0]);
+    };
+
+    fetchGreeting();
+  }, []);
```

Comment:
- According to the project requirements, this action should be performed in your redux store, please fix this 🙏. Please make sure your API calls are dispatched from the redux store and your app state is gotten from there instead of using the React useState hook.